### PR TITLE
feat: integrate WeaveID login

### PR DESCRIPTION
**Pull request by leNicDev**
Integrate WeaveID based on [https://community.xyz/opportunity#QfwCoWMZq-L1WZe580RpsOQLjWWq4iXGDW7Zvifi1_4](https://community.xyz/opportunity#QfwCoWMZq-L1WZe580RpsOQLjWWq4iXGDW7Zvifi1_4)

### CODE REVIEW

**Comment by t8** [comment]
Thanks for the contribution! I'm not sure if this is just me, but it seems that I'm unable to add another keyfile using this method after logging in for the first time. 

CC @martonlederer

**Comment by leNicDev** [comment]
@t8 I managed to reproduce it. I will start working on a fix later today!

**Comment by leNicDev** [comment]
Fixed a bug that caused the profiles to become `undefined` when the user tried logging in using the same key file twice.

File: src/routes/login.svelte [hunk 1]
```diff
@@ -131,6 +154,15 @@
         have a keyfile, you can get one by creating an <a
           href="https://www.arweave.org/wallet">Arweave Wallet</a>.
       </p>
+      <div class="weaveid-login">
```
Comment (t8) [inline src/routes/login.svelte:157]:
```suggestion
      <hr />
      <div class="weaveid-login">
```
I think we need some sort of differentiator so users know they can drag/drop their keyfile *or* login with WeaveID.

File: src/routes/login.svelte [hunk 2]
```diff
@@ -131,6 +154,15 @@
...
+        {#if !!weaveIdClient}
+          <button class="weave-id-button" on:click={openWeaveIdLoginModal}>
+            <img src={weaveidSVG} alt="weaveid" /> WeaveID
```
Comment (t8) [inline src/routes/login.svelte:160]:
```suggestion
            <img src={weaveidSVG} alt="weaveid" /> WeaveID Login
```

Comment (t8) [reply src/routes/login.svelte:160]: Would be interested in hearing @maximousblk's thoughts on this.

**Review by t8 — CHANGES_REQUESTED** [review]
We're so close! This is looking great, @leNicDev. Please address the other comments in this review and then we can merge. 

Also - I'm not sure if this is an us issue or a WeaveID issue, but the sequence of clicking the "WeaveID" button, exiting the modal, and re-clicking the button throws this error for me:
image

We may need to get in touch with Seena for this one.

File: src/routes/login.svelte (hunk 2 above)
Comment (maximousblk) [reply src/routes/login.svelte:160]: imo go with `WeaveID` or `use WeaveID`

Comment (t8) [reply src/routes/login.svelte:160]: Ahh, I like `use WeaveID`. Thanks!

File: src/routes/login.svelte (hunk 1 above)
Comment (leNicDev) [reply src/routes/login.svelte:157]: What do you think about this?
image

Comment (leNicDev) [reply src/routes/login.svelte:166]: @t8 The error you mentioned has been fixed.

**Comment by littledivy** [comment]
@leNicDev Was playing around with this and encountered an annoying behavior with the WeaveID modal. When trying to go out of the modal by clicking on the right side of the page the file selector is triggered.
ezgif com-video-to-gif(1)

File: src/routes/login.svelte (hunk 1 above)
Comment (t8) [reply src/routes/login.svelte:166]: This looks great! Thanks for addressing this.

**Comment by t8** [comment]
> @leNicDev Was playing around with this and encountered an annoying behavior with the WeaveID modal. When trying to go out of the modal by clicking on the right side of the page the file selector is triggered.
> ezgif com-video-to-gif(1)

I think this is an issue with the `z-index` of either WeaveID or our input. Any thoughts @martonlederer?

**Comment by leNicDev** [comment]
@littledivy @t8 Fixed it by increasing the z-index of `#popupdarkbg`. Might be a possible improvement for WeaveID.

**Review by t8 — APPROVED** [review]
LGTM. Amazing work @leNicDev!

**Review by martonlederer — APPROVED** [review]
Amazing work, thank you :D

**Review by johnletey — APPROVED** [review]
LGTM @leNicDev 😄

**Pull request merged**